sklearn/decomposition/tests/test_fastica.py

```diff
@@ -259,3 +259,15 @@ def test_inverse_transform():
             # reversibility test in non-reduction case
             if n_components == X.shape[1]:
                 assert_array_almost_equal(X, X2)
+
+
+def test_fastica_errors():
+    n_features = 3
+    n_samples = 10
+    rng = np.random.RandomState(0)
+    X = rng.random_sample((n_samples, n_features))
+    w_init = rng.randn(n_features + 1, n_features + 1)
+    assert_raises(ValueError, FastICA, max_iter=0)
```

Comment:
Nice that you added tests for other errors. Could you test test the error messages as well (this can happen that you get a ValueError not for the reason you are expecting, also it's always nice to check the error message e.g. that it contains the invalid value provided by the user)?

You can use assert_raises_regex or if you want to be amongst the cool kids `with pytest.raises(ValueError, match='some_regex')` if you want to be amongst the coolest pytest kids.